### PR TITLE
[Feat] Added implementation for saving docs on supabase

### PR DESCRIPTION
**Pull request by rafaelsideguide**
~- TODO: remove the comments on `log_job.ts` before deploying to prod~

### CODE REVIEW

File: apps/api/src/controllers/crawl-status.ts [hunk 1]
```diff
@@ -20,15 +21,27 @@ export async function crawlStatusController(req: Request, res: Response) {
...
-      current_step: current_step,
-      total: total,
-      data: job.returnvalue,
-      partial_data: partialDocs ?? [],
+      current,
+      current_url,
+      current_step,
+      total,
+      data: data ? data : null,
+      partial_data: jobStatus == 'completed' ? [] : partialDocs,
```
Comment (nickscamara) [inline apps/api/src/controllers/crawl-status.ts:44]: We lose the partial_data when docs are completed. We need partial data to be there as people rely on it.

**Comment by nickscamara** [comment]
When running with crawlerOptions.returnOnlyUrls, save job throws this error
```
Failed to update job status: Error: canceling statement due to statement timeout
    at saveJob (/Users/nicolascamara/Documents/GitHub/firecrawl/apps/api/src/main/runWebScraper.ts:120:24)
```

**Comment by nickscamara** [comment]
- [ ] need index on job id on firecrawl_jobs, it is a bit slow

**Comment by rafaelsideguide** [comment]
Index is done. Pagination will be implemented on a v1.

**Comment by nickscamara** [comment]
@rafaelsideguide what about the other issues above? partial data and returnOnlyUrls?

**Comment by mogery** [comment]
@nickscamara Added code here that uses Supabase as backing for jobs that are in progress when the server crashes/restarts. On exit, jobs that are active are force unlocked and saved to the DB with `retry: true`. On startup, DB entries with `retry: true` are loaded back into the queue with status `"waiting"`, so the workers will pick them back up.

We are currently not saving scrape progress or `partialDocs`, but it would be a logical next step. Unfortunately it would require some pretty big adjustments to the WebScraperDataProvider.

**Comment by rafaelsideguide** [comment]
I did a load testing (1753 requests) while deploying on staging server and it ended up with 306 stuck jobs in active status. This implementation is better than the one I did few months ago bc back then I had about 30% of jobs stuck, whereas now we're getting 17%, but it seems we're still missing something...

File: apps/api/src/controllers/crawl-status.ts (hunk 1 above)
Comment (nickscamara) [reply apps/api/src/controllers/crawl-status.ts:44]: Added to the docs that it will become empty when the jobs complete. It should be fine.

**Review by nickscamara — APPROVED** [review]
lgtm

**Pull request merged**